WickedEngine/RenderPath3D.cpp

```diff
@@ -402,8 +402,12 @@ void RenderPath3D::RenderReflections(CommandList cmd) const
 		}
 
 		wiRenderer::SetClipPlane(water, cmd);
-
+#ifdef _WIN32
 		device->Barrier(&GPUBarrier::Image(&rtReflection, IMAGE_LAYOUT_SHADER_RESOURCE, IMAGE_LAYOUT_RENDERTARGET), 1, cmd);
+#else
+		GPUBarrier image = GPUBarrier::Image(&rtReflection, IMAGE_LAYOUT_SHADER_RESOURCE, IMAGE_LAYOUT_RENDERTARGET);
```

Comment:
No idea if GPUBarrier has complex assignment operator but if it does then we can make this minimally more efficient by doing:
```cpp
GPUBarrier image(GPUBarrier::Image(&rtReflection, IMAGE_LAYOUT_SHADER_RESOURCE, IMAGE_LAYOUT_RENDERTARGET));
```

Reply:
That's definitely something to look into